modules/TargetSafety.py

```diff
@@ -0,0 +1,305 @@
+        F.trim(F.col('official_symbol')).alias('targetFromSourceId'),
+        F.col('biological_process_target').alias('event'),
+        'eventId',
+        F.struct(
+            F.col('tissue').alias('tissueLabel'),
+            F.lit(None).alias('tissueId'),
+            F.col('cell_short_name').alias('cellLabel'),
+            F.col('cell_format').alias('cellFormat'),
+            F.lit(None).alias('cellId'),
+        ).alias('biosample'),
+        F.lit('ToxCast').alias('datasource'),
+        F.lit('https://www.epa.gov/chemical-research/exploring-toxcast-data-downloadable-data').alias('url'),
+        F.struct(
+            F.col('assay_component_endpoint_name').alias('name'),
+            F.col('assay_component_desc').alias('description'),
+            F.col('assay_format_type').alias('type'),
+        ).alias('study'),
+    )
+
+
+def initialize_spark():
+    """Spins up a Spark session."""
+
+    # Initialize spark session
+    spark_mem_limit = detect_spark_memory_limit()
+    spark_conf = (
+        SparkConf()
+        .set('spark.driver.memory', f'{spark_mem_limit}g')
+        .set('spark.executor.memory', f'{spark_mem_limit}g')
+        .set('spark.driver.maxResultSize', '0')
+        .set('spark.debug.maxToStringFields', '2000')
+        .set('spark.sql.execution.arrow.maxRecordsPerBatch', '500000')
+    )
+    spark = SparkSession.builder.config(conf=spark_conf).master('local[*]').getOrCreate()
+    logging.info(f'Spark version: {spark.version}')
+
+    return spark
+
+def get_parser():
+    """Get parser object for script TargetSafety.py."""
+    parser = argparse.ArgumentParser(description=__doc__)
+
+    parser.add_argument(
+        '--toxcast',
+        help='Input table containing biological processes associated with relevant targets that have been observed in toxicity assays.',
+        type=str,
+        required=True,
+    )
+    parser.add_argument(
+        '--adverse_events',
+        help='Input TSV containing adverse events associated with targets that have been collected from relevant publications. Fetched from GitHub..',
+        type=str,
+        required=True,
+    )
+    parser.add_argument(
+        '--safety_risk', help='Input TSV containing cardiovascular safety liabilities associated with targets that have been collected from relevant publications. Fetched from GitHub.', type=str, required=True
+    )
+    parser.add_argument(
+        '--output', help='Output gzipped json file following the target safety liabilities data model.', type=str, required=True
+    )
+    parser.add_argument(
+        '--log_file',
+        help='Destination of the logs generated by this script. Defaults to None',
+        type=str,
+        nargs='?',
+        default=None
+    )
+
+    return parser
+
+if __name__ == '__main__':
+    args = get_parser().parse_args()
+    
+    main(
+        toxcast=args.toxcast,
+        adverse_events=args.adverse_events,
+        safety_risk=args.safety_risk,
+        output=args.output
```

Comment:
If the arguments are passed like this, the `safety_risk` will always be submitted even if the script was not called with safety risk file as being optional. This have a side effect: the default value defined in row 28,29. But if the other comments are sorted, this will not be an issue.